packages/jss/src/utils/escape.js

```diff
@@ -2,14 +2,12 @@ const env = process.env.NODE_ENV
 
 const escapeRegex = /([[\].#*$><+~=|^:(),"'`])/g
 
+const nativeEscape = typeof CSS !== 'undefined' && CSS.escape
+
 export default str => {
   // We don't need to escape it in production, because we are not using user's
   // input for selectors, we are generating a valid selector.
   if (env === 'production') return str
```

Comment:
Quick question about that: What happens if the user has a custom generate-id and passed some non-valid css selector.

Reply:
in that case its up to the user, because we want to have max perf, escaping has a cost

Reply:
We should maybe note that in the docs about a custom generate-id if it's not already there.

Reply:
yeah, the resulting id should be a selector conform string